lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -25,11 +25,27 @@ def build_index(self, server_config, dates):
             index += f"{idx}{d},"
         return index
 
-    def test_missing_json_object(self, client, server_config):
+    def test_non_accessible_user_data(self, client, server_config, pbench_token):
```

Comment:
There is a matrix, here:
requestor | target | result
-- | -- | --
valid (e.g., `drb`) | valid (e.g., `drb`) | ✅ 
valid (e.g., `drb`) | invalid (e.g., `pp`) | ❌ 
valid (e.g., `drb`) | public (e.g., `"user": ""`) | ✅ 
valid (e.g., `drb`) | public (e.g., `"user"` field omitted) | ✅ 
invalid (e.g., `pp`) | valid (e.g., `drb`) | ❌ 
invalid (e.g., `pp`) | invalid (e.g., `pp`) | ❌ 
invalid (e.g., `pp`) | public (e.g., `"user": ""`) | ❓ 
invalid (e.g., `pp`) | public (e.g., `"user"` field omitted) | ❓ 
none (e.g., no token) | valid (e.g., `drb`) | ❌ 
none (e.g., no token) | invalid (e.g., `pp`) | ❌ 
none (e.g., no token) | public (e.g., `"user": ""`) | ✅ 
none (e.g., no token) | public (e.g., `"user"` field omitted) | ✅ 

How many of these cases do we have tests for?  (We might be able to do this with a single, parameterized test; or we could do it with a few of them.)

(Re the ❓ entries:  I don't know whether these should succeed or fail -- we need to decide...do we treat an invalid user/token/login differently from no-user/token/login?)

Reply:
Most of these tests are present now but I have some confusion
requestor | target | result |Tested
-- | -- | -- | --
valid (e.g., `drb`) | valid (e.g., `drb`) | ✅ |✅
valid (e.g., `drb`) | invalid (e.g., `pp`) | ❌ |✅
valid (e.g., `drb`) | public (e.g., `"user": ""`) | ✅ |✅
valid (e.g., `drb`) | public (e.g., `"user"` field omitted) | ✅ |✅
invalid (e.g., `pp`) | valid (e.g., `drb`) | ❌ |✅ (vice-versa as 2nd case)
invalid (e.g., `pp`) | invalid (e.g., `pp`) | ❌ |(by invalid requester do you mean unauthenticated?)
invalid (e.g., `pp`) | public (e.g., `"user": ""`) | ❓| =
invalid (e.g., `pp`) | public (e.g., `"user"` field omitted) | ❓|=
none (e.g., no token) | valid (e.g., `drb`) | ❌ |(Does invalid and unauthenticated user mean the same thing?)
none (e.g., no token) | invalid (e.g., `pp`) | ❌| =
none (e.g., no token) | public (e.g., `"user": ""`) | ✅ |=
none (e.g., no token) | public (e.g., `"user"` field omitted) | ✅ |=

Reply:
> by invalid requester do you mean unauthenticated?

> Does invalid and unauthenticated user mean the same thing?

By "invalid", I think what I meant was a request specifying a token which is invalid.  I believe that this case is distinct from a request which specifies no token (and, obviously, it's distinct from a request specifying a valid token).

Reply:
Got it, added more unittests for couple of missing scenarios from the table

Reply:
I believe this case is already covered in `test_non_accessible_user_data`

Reply:
Cool.  (Do we have a test which shows that administrative access succeeds?  If not, no biggie -- it's early days for that, yet.)

Reply:
No the test does not check for the admin access scenario, but that can be handled in another PR. I believe for all the tests files we need to add this admin access test since we now have the cli implemented.

Reply:
Anything that we discover during review that we're not going to address in the pull request should probably have an issue opened to track it.

Reply:
Created https://github.com/distributed-system-analysis/pbench/issues/2361